src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Windows/Markup/DateTimeValueSerializer.cs

```diff
@@ -50,29 +47,28 @@ public override object ConvertFromString(string value, IValueSerializerContext?
                 return DateTime.MinValue;
             }
 
-            // Get a DateTimeFormatInfo and set the formatting style for round-tripping
-            // and to trim the string.
-            DateTimeFormatInfo dateTimeFormatInfo = (DateTimeFormatInfo)TypeConverterHelper.InvariantEnglishUS.GetFormat(typeof(DateTimeFormatInfo))!;
+            // Set the formatting style for round-tripping and to trim the string.
             const DateTimeStyles DateTimeStyles = DateTimeStyles.RoundtripKind
-                      | DateTimeStyles.NoCurrentDateDefault
-                      | DateTimeStyles.AllowLeadingWhite
-                      | DateTimeStyles.AllowTrailingWhite;
-            return DateTime.Parse(value, dateTimeFormatInfo, DateTimeStyles);
+                                                | DateTimeStyles.NoCurrentDateDefault
+                                                | DateTimeStyles.AllowLeadingWhite
+                                                | DateTimeStyles.AllowTrailingWhite;
+            return DateTime.Parse(value, DateTimeFormatInfo.InvariantInfo, DateTimeStyles);
         }
 
 
         /// <summary>
-        /// Converts the given value object to a <see cref="T:System.DateTime"></see> using the arguments.
+        /// Converts the given value object to a <see cref="DateTime" /> using the arguments.
         /// </summary>
         public override string ConvertToString(object? value, IValueSerializerContext? context)
         {
-            if (value == null || !(value is DateTime dateTime))
+            if (value is not DateTime dateTime)
             {
                 throw GetConvertToException(value, typeof(string));
             }
 
             // Build up the format string to be used in DateTime.ToString()
-            var formatString = new StringBuilder("yyyy-MM-dd");
+            DefaultInterpolatedStringHandler formatSpan = new(5, 0, CultureInfo.CurrentCulture, stackalloc char[38]);
```

Comment:
Can I know why the number of char is 38?

Reply:
It should have been 36 actually.